draft-packages/menu/KaizenDraft/Menu/Menu.tsx

```diff
@@ -1,84 +1,90 @@
 import { ButtonProps } from "@kaizen/draft-button"
-import * as React from "react"
+import React, { useState } from "react"
 import MenuDropdown from "./MenuDropdown"
 
 const styles = require("./styles.scss")
 
-type MenuState = {
-  isMenuVisible: boolean
-}
-
-export type MenuProps = {
+export type GenericMenuProps = {
   align?: "left" | "right"
-  button: React.ReactElement<ButtonProps>
   menuVisible?: boolean
   automationId?: string
+  children: React.ReactNode
 }
 
-export default class Menu extends React.Component<MenuProps, MenuState> {
-  static displayName = "Menu"
-  static defaultProps = {
-    iconPosition: "start",
-    align: "left",
-  }
+type StatefulMenuProps = {
+  button: React.ReactElement<ButtonProps>
+}
 
-  dropdownButtonContainer = React.createRef<HTMLDivElement>()
+export type MenuProps = GenericMenuProps & StatefulMenuProps
 
-  constructor(props: MenuProps) {
-    super(props)
+type Menu = React.FunctionComponent<MenuProps>
 
-    this.state = {
-      isMenuVisible: Boolean(props.menuVisible),
-    }
-  }
+const Menu: Menu = props => {
+  const { align = "left", menuVisible = false } = props
 
-  toggleMenu = (e: MouseEvent) => {
-    e.stopPropagation()
+  const dropdownButtonContainer = React.createRef<HTMLDivElement>()
 
-    const currentState = this.state.isMenuVisible
-    this.setState({
-      isMenuVisible: !currentState,
-    })
-  }
+  const [isMenuVisible, setIsMenuVisible] = useState<boolean>(menuVisible)
 
-  hideMenu = () => {
-    this.setState({
-      isMenuVisible: false,
-    })
+  const toggleMenuDropdown = () => {
+    setIsMenuVisible(!isMenuVisible)
   }
 
-  getPosition() {
-    return this.dropdownButtonContainer && this.dropdownButtonContainer.current
-      ? this.dropdownButtonContainer.current.getBoundingClientRect()
-      : null
+  const hideMenuDropdown = () => {
+    setIsMenuVisible(false)
   }
 
-  renderMenuDropdown() {
-    return (
-      <MenuDropdown
-        hideMenuDropdown={this.hideMenu}
-        position={this.getPosition()}
-        align={this.props.align}
-      >
-        {this.props.children}
-      </MenuDropdown>
-    )
-  }
+  const { automationId, button, children } = props
 
-  render() {
-    const { automationId, button } = this.props
-    return (
-      <div
-        className={styles.dropdown}
-        data-automation-id={automationId}
-        ref={this.dropdownButtonContainer}
-      >
-        {React.cloneElement(button, {
-          onClick: this.toggleMenu,
-          onMouseDown: e => e.preventDefault(),
-        })}
-        {this.state.isMenuVisible && this.renderMenuDropdown()}
-      </div>
-    )
-  }
+  return render({
+    ...props,
+    align,
+    isMenuVisible,
+    dropdownButtonContainer,
+    hideMenuDropdown,
+    menuButton: React.cloneElement(button, {
+      onClick: e => {
+        e.preventDefault()
+        toggleMenuDropdown()
+      },
+      onMouseDown: e => e.preventDefault(),
+    }),
+  })
 }
+
+type RenderProps = {
```

Comment:
Couldn't think of a better name for this type. It's the props for the `render` function - nothing to do with the "render prop" pattern.